sphinx/util/inspect.py

```diff
@@ -484,7 +484,17 @@ def signature(subject: Callable, bound_method: bool = False, follow_wrapped: boo
             if len(parameters) > 0:
                 parameters.pop(0)
 
-    return inspect.Signature(parameters, return_annotation=return_annotation)
+    try:
+        # To allow to create signature object correctly for pure python functions,
+        # pass an internal parameter __validate_parameters__=False to Signature
+        #
+        # For example, this helps a function having a default value `inspect._empty`.
+        # refs: https://github.com/sphinx-doc/sphinx/issues/7935
+        return inspect.Signature(parameters,  # type: ignore
+                                 return_annotation=return_annotation,
+                                 __validate_parameters__=False)
+    except Exception:
+        return inspect.Signature(parameters, return_annotation=return_annotation)
```

Comment:
It would be better to add a comment here why this block is needed.

Reply:
I added this block for a fallback when target python does not support `__validate_parameters__` option. But it would be enough even after new python dropped the option. So I drop this except-clause now.